### PR TITLE
Caching and interval limit

### PR DESCRIPTION
**Pull request by Ezibenroc**
The aim of this PR is to finally be able to draw the plot for a large time interval. We were limited to an interval of 16 weeks, and 10000 queries.

The PR does two things : 
* It removes the limit on the length of the time interval. You can now plot for 10 years.
* It implements a very simple cache. The aim is to avoid downloading all the data at each time you want to plot something.

Then, I would like to increase the limit of 10000 queries. Or even better, remove this limit.
The caching will help to avoid downloading too much data from the server.

### CODE REVIEW

File: requests_graph.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def human_to_seconds(string):
 GRANULOMETRY = human_to_seconds(args.granulometry)
 TICKS = args.ticks
 LOGGER_URL = args.logger_url
+DATA_CACHE_FILENAME = 'requests_cache.json'
```
Comment (ProgVal) [inline requests_graph.py]: Could you make this configurable using a command-line argument?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,3 @@
 *pickle
 *.png
-*~
```
Comment (ProgVal) [inline .gitignore]: Why did you remove this line?

**Comment by Ezibenroc** [comment]
I made the file name configurable, and corrected the gitignore (it was a mistake).

I am wondering if json is the best format to cache data like this (for instance, we could use pickle). What do you think?

**Comment by ProgVal** [comment]
> I am wondering if json is the best format to cache data like this (for instance, we could use pickle). What do you think?

It keeps things simple. pickle would work too and is faster, though.

The “right” way would be to use rrdtool or a similar library, but I
don't think we need that kind of performance.

**Comment by Ezibenroc** [comment]
Alright, so let's keep json.

**Comment by ProgVal** [comment]
I did not notice it before, but the cache neither expires or is renewed. And it cannot be disabled.
